dev-tools/mage/check.go

```diff
@@ -301,3 +318,19 @@ func checkTitle(re *regexp.Regexp, title string, module string) error {
 	}
 	return nil
 }
+
+func checkDashboardIndexPattern(expectedIndex string, o *dashboardObject) error {
+	if objectMeta := o.Attributes.KibanaSavedObjectMeta; objectMeta != nil {
+		if index := objectMeta.SearchSourceJSON.Index; index != "" && index != expectedIndex {
+			return errors.Errorf("unexpected index pattern reference found in object meta: %s", index)
+		}
+	}
+	if visState := o.Attributes.VisState; visState != nil {
+		for _, control := range visState.Params.Controls {
+			if index := control.IndexPattern; index != "" && index != expectedIndex {
```

Comment:
When index will be ""?

Reply:
The field is optional, when it is not set it would be `""` here.

Reply:
oh I didn't know it's optional 😬 Thanks!